### PR TITLE
Deprecate node_side() API

### PR DESCRIPTION
**Pull request by tangxifan**
This is the final PR in the PR series #1667 and #1661  to replace node_side() API with is_node_on_specific_side() APIs

#### Description

- [X] Deprecate node_side() API
- [x] Deprecate node_sides() API

#### Related Issue

To address issue #1577 

#### Motivation and Context

To address issue #1577 

#### How Has This Been Tested?

Should have no impact to existing codes as the node_side() API is almost unused.

#### Types of changes

- [X] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: vpr/src/route/rr_node.cpp [hunk 1]
```diff
@@ -25,7 +25,13 @@ const char* t_rr_node::direction_string() const {
 }
 
 const char* t_rr_node::side_string() const {
```
Comment (litghost) [inline vpr/src/route/rr_node.cpp:27]: Probably need to deprecate this one too?

Comment (vaughnbetz) [reply vpr/src/route/rr_node.cpp:27]: Seems reasonable to deprecate.

Comment (tangxifan) [reply vpr/src/route/rr_node.cpp]: @litghost @vaughnbetz The method has been deprecated

**Comment by vaughnbetz** [comment]
Thanks Xifan; looks good.  Will merge as soon as CI goes green.

**Pull request merged**